crates/ty_server/src/server.rs

```diff
@@ -68,6 +55,20 @@ impl Server {
             global_settings.tracing.log_file.as_deref(),
         );
 
+        let client_capabilities = init_params.capabilities;
+        let position_encoding = Self::find_best_position_encoding(&client_capabilities);
+        let server_capabilities =
+            Self::server_capabilities(position_encoding, global_settings.experimental.as_ref());
+
+        let connection = connection.initialize_finish(
+            id,
+            &server_capabilities,
+            crate::SERVER_NAME,
+            crate::version(),
+        )?;
+
+        crate::message::init_messenger(connection.make_sender());
```

Comment:
I think it's important that `init_logging` (line 53) happens *after* `init_messenger`

Reply:
Yeah, I think so too. The messenger needs to be initialized first because it's being used when serializing the initialization options. If there are invalid initialization options, it requires the messenger to provide a notification to the user.